### PR TITLE
add new login style

### PR DESCRIPTION
**Pull request by theScrabi**
This is the implementation of https://github.com/owncloud/android/issues/2219

_____

**BUGS & IMPROVEMENTS**

- [x] (1) Avoid background resizing when opening the keyboard [WON'T BE FIXED]
- [X] (2) Branding https://github.com/owncloud/android/pull/2235#issuecomment-409601640
- [x] (3) Arrow in URL https://github.com/owncloud/android/pull/2235#issuecomment-411478878 [DONE] @jesmrec  
- [x] (4) Logo is cut in credentials mode https://github.com/owncloud/android/pull/2235#issuecomment-414738777 [WON'T BE FIXED]
- [x] (5) Background color https://github.com/owncloud/android/pull/2235#issuecomment-414944304 [FIXED]

### CODE REVIEW

File: res/values/styles.xml [hunk 1]
```diff
@@ -246,4 +246,27 @@
         <item name="android:textSize">16sp</item>
     </style>
 
+    <!-- Authenticator Activitu -->
```
Comment (jesmrec) [inline res/values/styles.xml]: Activity

File: res/values/styles.xml [hunk 2]
```diff
@@ -246,4 +246,27 @@
         <item name="android:textSize">16sp</item>
     </style>
 
+    <!-- Authenticator Activity -->
+    <style name="Theme.ownCloud.Authenticator"
+        parent="Theme.ownCloud">
+
```
Comment (davigonz) [inline res/values/styles.xml]: Too much space here

File: res/values/styles.xml [hunk 3]
```diff
@@ -246,4 +246,27 @@
...
+        <!-- properties for action mode -->
+        <item name="windowActionModeOverlay">true</item>
+        <item name="actionModeBackground">@color/action_mode_background</item>
+
```
Comment (davigonz) [inline res/values/styles.xml]: Same here

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 4]
```diff
@@ -1253,10 +1253,15 @@ private void onGetServerInfoFinish(RemoteOperationResult result) {
             mServerIsValid = false;
         }
 
+
+
+
         if (mServerIsValid) {
             updateAuthenticationPreFragmentVisibility();
         }
```
Comment (davigonz) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Same here

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 5]
```diff
@@ -1253,10 +1253,15 @@ private void onGetServerInfoFinish(RemoteOperationResult result) {
             mServerIsValid = false;
         }
 
+
```
Comment (davigonz) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Same here

File: res/layout-land/account_setup.xml [hunk 6]
```diff
@@ -17,203 +17,214 @@
...
+            <ScrollView
+                android:id="@+id/scroll"
+                android:layout_width="0dp"
+                android:layout_height="match_parent"
+                android:layout_weight="1"
+                android:fillViewport="true"
+                android:orientation="vertical">
+
+                <LinearLayout
+                    android:id="@+id/LinearLayout1"
```
Comment (davigonz) [inline res/layout-land/account_setup.xml]: Please use a more descriptive name for this linear layout

File: res/layout-land/account_setup.xml [hunk 7]
```diff
@@ -17,203 +17,214 @@
...
+                            android:layout_height="wrap_content"
+                            android:layout_gravity="bottom"
+                            android:contentDescription="@string/auth_host_address"
+                            android:drawablePadding="5dp"
+                            android:hint="@string/auth_host_url"
+                            android:inputType="textUri"
+                            android:paddingRight="55dp"
+                            style="@style/EditText.Authenticator"
+                            android:theme="@style/EditText.Authenticator">
+
```
Comment (davigonz) [inline res/layout-land/account_setup.xml]: Too much space here

**Comment by davigonz** [comment]
Would it be possible to distribute better the white spaces as the original design? https://github.com/owncloud/android/issues/2219

[](https://user-images.githubusercontent.com/15261363/42682430-e0175b48-868a-11e8-9321-14519117fc88.png)

**Comment by davigonz** [comment]
I like so much the new design but I miss a bit of intuitivity here and this problem comes from previous versions. 

I'm talking about the connect button, which appears as soon as the login view is opened and is useless till the username and password fields come out. 

Think we could use the same button but with different names:

- **Check server** to click on it after introducing the server address.
- **Log in** to click on it after introducing username and password.

**Comment by jesmrec** [comment]
My choice:

- **Connect** in the first
- **Log in** or **Authorize** in the second (same wording used in OAuth2)

Take in account that the first step behaves a bit different depending on the authentication method. For example, in OAuth2, when you tap the button you are performing more than a "check". 

as reference: iOS-neo uses "Continue" to move forward to the following step (URL -> credentials -> log in)

In any case, we can face this UX issue in a separated discussion.

**Comment by jesmrec** [comment]
### (2)

On the other side, the current feature seems to have branding concerns. At least, moving the changes in `setup.xml`.

**Comment by davigonz** [comment]
About what was commented in https://github.com/owncloud/android/pull/2235#issuecomment-409293117 and https://github.com/owncloud/android/pull/2235#issuecomment-409599937, we could do something like the gif below:

[](https://user-images.githubusercontent.com/15261363/43789705-e32574a6-9a70-11e8-9fff-6a0d896a196d.gif)

Here we have hidden the `Connect button` till the server is properly checked.

What do you think? @jesmrec @michaelstingl @pmaier1

**Comment by michaelstingl** [comment]
@davigonz What does trigger the server check? Every keystroke? What happens if I have `demo.owncloud.org` AND `demo.owncloud.org/testnextversion`? How can I connect to the 2nd one?

**Comment by davigonz** [comment]
> What does trigger the server check? Every keystroke?

@michaelstingl No, the server check will start after pressing any of the parts highlighted in the screenshot below. The most used option would likely be the **Next** button in the keyboard.

[](https://user-images.githubusercontent.com/15261363/43832701-8eec4764-9b08-11e8-8dae-62f2c4826c84.png)

> What happens if I have demo.owncloud.org AND demo.owncloud.org/testnextversion? How can I connect to the 2nd one?

You can repeat the same steps to connect the 2nd one

[](https://user-images.githubusercontent.com/15261363/43833000-878af12c-9b09-11e8-9ffa-097789321d36.gif)

**Comment by davigonz** [comment]
> On the other side, the current feature seems to have branding concerns. At least, moving the changes in setup.xml.

@jesmrec done

**Comment by michaelstingl** [comment]
hm, okay. let's try this… (can we test with a public beta?)

@pmaier1 @voroyam @cdamken @patrickjahns @pako81 How does this button-less behaviour feel for you as everyday Android users?

**Comment by voroyam** [comment]
From my pow:

There are no apps to my knowledge that behave the same - button-less. So it would be unexpected to look for buttons on the screen and then have to find it on the key board.

Also this is the login screen / connection screen. I have seen this once I configured my new phone and never again, so I don't think I can provide important feedback aside from what I already wrote.

**Comment by jesmrec** [comment]
@davigonz does it affect to credential edition mode?

**Comment by theScrabi** [comment]
Well there are apps similar to our one that show a little arrow on the end of the eddit field. This arroe behaves as the connect button.

**Comment by davigonz** [comment]
> There are no apps to my knowledge that behave the same - button-less. So it would be unexpected to look for buttons on the screen and then have to find it on the key board.

Thank you @voroyam for your opinion, I agree with you but the previous login behaviour was even worse in my opinion, since there was a "Connect" button that could not be used to check the server and was visible, making the users press on it and see that nothing happens. 

> Well there are apps similar to our one that show a little arrow on the end of the eddit field. This arrow behaves as the connect button.

Let's try with something like this

[](https://user-images.githubusercontent.com/15261363/43852329-3c640dee-9b3d-11e8-9354-88de6cacb69d.gif)

**Comment by jesmrec** [comment]
I think that the arrow does not add more value to the view, since the user can tap everywhere in the screen, apart of the key in the keyboard.  The approach here: https://github.com/owncloud/android/pull/2235#issuecomment-411121913 is the one that straightforward the user during the process in my opinion.

**Comment by davigonz** [comment]
@jesmrec all the login icons are already prepared to be brandeable, so what is left is whether we use the arrow in the server address field or not.

**Comment by jesmrec** [comment]
Regarding the use of the arrow to confirm the typed URL, more thoughts are welcome: @michaelstingl @sd1998 @pmaier1 ...

**Comment by michaelstingl** [comment]
@jesmrec let's try this…

**Comment by jesmrec** [comment]
@michaelstingl you mean with or without the arrow

**Comment by michaelstingl** [comment]
Let's try the arrow…

**Comment by jesmrec** [comment]
ok, we will include it also for the beta.

**Comment by jesmrec** [comment]
### (4)

1. Enter correct URL
2. When credentials fields are displayed (soft keyboard), scroll the view

Current: ownCloud logo is cut
Expected: ownCloud logo is not cut:


aug-21-2018 18-29-37 1

**Comment by jesmrec** [comment]
### (5)

Suggestion: with the changes in `setup.xml` i suggest to set:

`<color name="login_background_color">@color/owncloud_blue</color>`

instead of 

`<color name="login_background_color">@color/white</color>`

because in case of `false` in the `use_login_background_image` parameter, the login view is totally white.

**Comment by theScrabi** [comment]
I think the alternative for [(4)](https://github.com/owncloud/android/pull/2235#issuecomment-414738777) would be to remove the could once the keyboard is shown, however that's rather weird.

I see it like this. If you get to the login screen for the first time, you see all including the cloud. But once you tab into a filed the keyboard shows up. here it's important to enter your credentials, not to have the cloud fully visible. By letting it "cut" you may also show the user that he can scroll the layout.

**Comment by davigonz** [comment]
> If you get to the login screen for the first time you see all, including the cloud, but once you tab into a filed the keyboard shows up. here it's important to enter your credentials, not to have the cloud fully visible. By letting it "cut" you may also show the user that he can scroll the layout.

Agree with this, for me is not a bad behaviour.

**Comment by jesmrec** [comment]
@theScrabi of course, it is not an important issue because in the moment the keyboard raises, the focus is on credentials and not in background colors and images. But take in mind that we should not include regression issues in the app, and all suggestions and behaviours are susceptible to be reported regardless they will or won't be fixed.

**Comment by davigonz** [comment]
@jesmrec bugs and improvements done

**Comment by jesmrec** [comment]
all stuff fixed. Approved.

**Comment by davigonz** [comment]
To be included in 2.9 version

**Pull request merged**